Robust.Shared/Prototypes/PrototypeManager.cs

```diff
@@ -904,4 +916,12 @@ private void OnReload(PrototypesReloadedEventArgs args)
             return data;
         }
     }
+
+    [Virtual]
+    public class InvalidPrototypeNameException : Exception
```

Comment:
[Virtual] is in case you intend for this class to be inherited. I don't think there's much of a reason to let content inherit this exception type.

```suggestion
    public sealed class InvalidPrototypeNameException : Exception
```

Reply:
Oh I just copied it from component. InvalidComponentName exception is virtual, perhaps it makes sense to just inherit it?

Reply:
The two should be separate though I don't know why the component one is virtual either